### PR TITLE
fix: replace stakingplus module with staking hooks

### PR DESCRIPTION
**Pull request by ulbqb**
## Description

This PR replaces stakingplus module with staking hooks. It makes redundant files removed and maintenance easy.

This change has breaking change of proto.

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Comment by ulbqb** [comment]
I have an issue.
Should stakingplus proto file be modified? This PR makes modification to be minimal files. It's also possible to use proto file without modification. But if so, it contains redundant files.

**Comment by tkxkd0159** [comment]
> I have an issue. Should stakingplus proto file be modified? This PR makes modification to be minimal files. It's also possible to use proto file without modification. But if so, it contains redundant files.

I think current status looks not bad.

**Review by tkxkd0159 — DISMISSED** [review]

**Review by jaeseung-bae — COMMENTED** [review]
Is there any test for hooks to work fine?
I think it's also fine if there is any test in upstream.

File: x/foundation/keeper/internal/hooks.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	return nil
+}
+
+func (h Hooks) AfterValidatorCreated(goCtx context.Context, valAddr sdk.ValAddress) error {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	grantee := sdk.AccAddress(valAddr)
+
+	msg := stakingtypes.MsgCreateValidator{
+		ValidatorAddress: valAddr.String(),
```
[the remark this answers is not in the document]
Comment (ulbqb) [reply x/foundation/keeper/internal/hooks.go:44]: Which codec should be used, staking or foundation?

**Review by jaeseung-bae — APPROVED** [review]

**Review by 0Tech — APPROVED** [review]

**Pull request merged**